src/Common/src/Interop/Unix/System.Native/Interop.GetNativeIPInterfaceIndex.cs

```diff
@@ -0,0 +1,15 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetNativeIPInterfaceIndex")]
```

Comment:
This should use SetLastError = true, since the method uses errno for errors.

Reply:
It's just a light-weight copy of https://github.com/dotnet/corefx/blob/master/src/Common/src/Interop/BSD/System.Native/Interop.ProtocolStatistics.cs#L159-L160 - should I add SetLastError there too?

Reply:
Ideally yes, although it looks like the call sites don't currently pay attention to or incorporate errno in their thrown exceptions.